### PR TITLE
 build/linter/util: exclude files according to the key of `only_files`

### LINKED ISSUE #48998: `only_files` configuration in linter doesn't work properly
`linter/util/exclude.go` excludes files according to the values of `only_files`, but it's expected to use the keys.

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Issue Number: close #48998

Problem Summary:

The config use `key` to represent the pattern, but not value. I copied the code from [`nogo_main.go`](https://github.com/bazelbuild/rules_go/blob/master/go/tools/builders/nogo_main.go), but in `nogo_main` it's an array :facepalm: .

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: build/linter/util/exclude_test.go [hunk 1]
```diff
@@ -23,4 +23,5 @@ import (
 func TestShouldRun(t *testing.T) {
 	require.True(t, shouldRun("gofmt", "some.go"))
 	require.False(t, shouldRun("gofmt", "uca_generated.go"))
+	require.True(t, shouldRun("revive", "/pkg/meta/distributed_lock.go"))
```
Comment (tangenta) [inline build/linter/util/exclude_test.go:26]: The PR that creates this file is not merged yet. #48971
```suggestion
	require.True(t, shouldRun("revive", "/pkg/meta/meta.go"))
```

Comment (YangKeao) [reply build/linter/util/exclude_test.go:26]: No problem. This test only uses the string, but not the file content.

**Review by tangenta — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
lgtm

**Pull request merged**